app/src/main/java/org/mozilla/fenix/home/collections/CollectionViewHolder.kt

```diff
@@ -0,0 +1,103 @@
+ * Clients are expected to use [bindSession] to link a particular [TabCollection] to be displayed
+ * otherwise this will be an empty, 0 size View.
+ *
+ * @param composeView [ComposeView] which will be populated with Jetpack Compose UI content.
+ * @param viewLifecycleOwner [LifecycleOwner] to which this Composable will be tied to.
+ * @param interactor [CollectionInteractor] callback for user interactions.
+ */
+class CollectionViewHolder(
+    composeView: ComposeView,
+    viewLifecycleOwner: LifecycleOwner,
+    private val interactor: CollectionInteractor,
+) : ComposeViewHolder(composeView, viewLifecycleOwner) {
+    private var collectionData = CollectionInfo()
+
+    @Composable
+    override fun Content() {
+        val collectionInfo by remember { mutableStateOf(collectionData) }
+
+        collectionInfo.collection?.let { validCollection ->
+            val horizontalPadding =
+                composeView.resources.getDimensionPixelSize(R.dimen.home_item_horizontal_margin)
+            composeView.setPadding(horizontalPadding, 0, horizontalPadding, 0)
+
+            val menuItems = getMenuItems(
+                collection = validCollection,
+                onCollectionOpenTabsTapped = interactor::onCollectionOpenTabsTapped,
+                onRenameCollectionTapped = interactor::onRenameCollectionTapped,
+                onCollectionAddTabTapped = interactor::onCollectionAddTabTapped,
+                onDeleteCollectionTapped = interactor::onDeleteCollectionTapped,
+            )
+
+            Column {
+                Spacer(Modifier.height(12.dp))
+
+                Collection(
+                    collection = validCollection,
+                    expanded = collectionInfo.isExpanded,
+                    menuItems = menuItems,
+                    onToggleCollectionExpanded = interactor::onToggleCollectionExpanded,
+                    onCollectionShareTabsClicked = interactor::onCollectionShareTabsClicked,
+                    onCollectionMenuOpened = interactor::onCollectionMenuOpened,
+                )
+            }
+        }
+    }
+
+    /**
+     * Dynamically replace the current [TabCollection] shown in this `ViewHolder`.
+     *
+     * @param collection [TabCollection] to be shown
+     * @param expanded Whether to show the collection as expanded or collapsed.
+     */
+    fun bindSession(collection: TabCollection, expanded: Boolean) {
+        collectionData = CollectionInfo(
+            collection = collection,
+            isExpanded = expanded
+        )
+    }
+
+    companion object {
+        val LAYOUT_ID = View.generateViewId()
+        const val maxTitleLength = 20
+    }
+}
+
+/**
+ * Wrapper over a [TabCollection] adding information about whether it should be shown as expanded or collapsed.
+ *
+ * @property collection [TabCollection] to display.
+ * @property isExpanded Whether the collection is expanded to show it's containing tabs or not.
+ */
+@Stable
+private data class CollectionInfo(
```

Comment:
Can this just be `Collection`?

Reply:
Trying this rename I get
`Class 'CollectionInfo' will be shadowed by function 'Collection'`
as the Composable showing the collection already has this name.
There are quite a few `collection` related constructs. In this case `CollectionInfo` I think helps express a bit more about what it represents and help in knowing how and where to use it.

Reply:
Ok. I was a bit effy about the `Info` suffix since we didn't any data classes that ended in that suffix in Fenix compare to `VisitInfo` in AC.